### ISSUE TITLE
Kodi Connection Error

### ISSUE THREAD

**Issue by DarkHunterer** [hyperion-project/hyperion#581]
I use raspberry B+ with OSMC + hyperion. LEDS are controlled with ws2811 via arduino.

When I use sudo service hyperion status this is what I get 

osmc@osmc:~$ sudo service hyperion status
● hyperion.service - Hyperion Systemd service
   Loaded: loaded (/etc/systemd/system/hyperion.service; enabled)
   Active: active (running) since Fri 2016-04-01 21:24:58 CEST; 6min ago
 Main PID: 757 (hyperiond)
   CGroup: /system.slice/hyperion.service
           └─757 /opt/hyperion/bin/hyperiond /etc/hyperion.config.json

Apr 01 21:30:19 osmc hyperiond[757]: KODICHECK ERROR: Kodi Connection error (0)
Apr 01 21:30:29 osmc hyperiond[757]: KODICHECK ERROR: Kodi Connection error (0)
Apr 01 21:30:39 osmc hyperiond[757]: KODICHECK ERROR: Kodi Connection error (0)
Apr 01 21:30:49 osmc hyperiond[757]: KODICHECK ERROR: Kodi Connection error (0)
Apr 01 21:30:59 osmc hyperiond[757]: KODICHECK ERROR: Kodi Connection error (0)
Apr 01 21:31:09 osmc hyperiond[757]: KODICHECK ERROR: Kodi Connection error (0)
Apr 01 21:31:19 osmc hyperiond[757]: KODICHECK ERROR: Kodi Connection error (0)
Apr 01 21:31:29 osmc hyperiond[757]: KODICHECK ERROR: Kodi Connection error (0)
Apr 01 21:31:39 osmc hyperiond[757]: KODICHECK ERROR: Kodi Connection error (0)
Apr 01 21:31:49 osmc hyperiond[757]: KODICHECK ERROR: Kodi Connection error (0)
 
I can't use hyperion because of ( I suppose) this error. Any help, any ideas please?

Hyperion config file
http://pastebin.com/Npnnj3E3

**Comment by brindosch** [comment]
In Kodi
Enable "Allow remote control by programs on this system"
and "Allow remote control by programs on other systems" (you need both, even on the same machine)

navigate ->System->Services->Remote control

**Comment by brindosch** [comment]
@DarkHunterer 
News?

**Comment by DarkHunterer** [comment]
I managed to get rid of that error before the first answer, so I doubt if that was the cause.

I think I just removed hyperion and run install script again.

Howerver, I found 2 more things:

1) I was cursing, because I couldn't make hyperion work. It turned out that baudrate I used was propably too high (460800). So this is propably not because of hyperion, just raspberry/arduino stuff.

2) I use baudrate 115200 now and it looks like its working perfectly. However, when I run rgb test video, entire screen of color looks decent but when there is something like screen divided to 4 pieces with diagonals, the colors are not right. The "left" color overlaps most of the leds, like it is the most important. So it looks like it is not calibrated right, any help with that?

**Comment by brindosch** [comment]
It seams you have to adjust the colors. Did you calibrated it before? Play around with color settings maybe that helps

**Comment by DarkHunterer** [comment]
This doesn't like color adjustment problem for me.
http://postimg.org/image/kxkvwwd3h/f32ea702/

I would say that the colors are 90 degrees ahead from what it should be. 

Any solution for that?

**Comment by brindosch** [comment]
https://github.com/tvdzwan/hyperion/wiki/FAQ-Frequently-Asked-(Problems)-Questions
Point 3, if not may be 5

**Comment by DarkHunterer** [comment]
@brindosch Thank you for you answers.

Well, of course I saw FAQ and played with HyperCon MANY times and I think it can be fault of HyperCon. 

This is view of my settings, including offset http://imgur.com/82NXAQE

What I noticed is that left color is the most "important". It overlaps the entire top very fast, and the right color is very easly overlaped by the top.

Can you recommend something? Did you change something in Hyperion and forgot to refresh HyperCon maybe?

**Comment by brindosch** [comment]
hm, no and no. What happen if you play the red dot video, is the position at the right place?
http://insaneboard.de/blog/wp-content/uploads/2014/10/RedDot-720p.zip

**Comment by DarkHunterer** [comment]
Great example. It shows the "power" of the left side.

And here is the effect
[V_20160406_230054.mp4.zip](https://github.com/tvdzwan/hyperion/files/207352/V_20160406_230054.mp4.zip)

**Comment by brindosch** [comment]
wow, completely screwed. I will have a look.

**Comment by DarkHunterer** [comment]
Thank you for your time, let me know.

**Comment by brindosch** [comment]
How big is your grabbing are of your frame gabber?

**Comment by DarkHunterer** [comment]
"framegrabber" :
        {
                "width" : 80,
                "height" : 45,
                "frequency_Hz" : 10.0,
                "priority" : 890
        },

I think this is it? I didn't change anything, so I think these are defaults

**Comment by brindosch** [comment]
The new Hyperion version contains a new driver for ws2811 for gpio direct connection, could you test this? I don´t get it.

**Comment by tpmodding** [comment]
1. can you try this

"framegrabber" :
{
"width" : 720,
"height" : 576,
"frequency_Hz" : 10.0,
"priority" : 890
},
2. try another direction...

just test this, and than we'll see

**Comment by DarkHunterer** [comment]
@tpmodding it didn't really change anything. And another direction is well... another direction.
@brindosch So what I did is : I used HyperCon with "ws281x" type. It's the one, right? I assume that it gives output on gpio18. However, when I upload file with settings for ws281x there is this output...


● hyperion.service - Hyperion Systemd service
   Loaded: loaded (/etc/systemd/system/hyperion.service; enabled)
   Active: failed (Result: start-limit) since Fri 2016-04-08 21:22:08 CEST; 7s ago
  Process: 652 ExecStart=/opt/hyperion/bin/hyperiond /etc/hyperion.config.json (code=killed, signal=SEGV)
 Main PID: 652 (code=killed, signal=SEGV)

Apr 08 21:22:07 osmc hyperiond[652]: "leds" : 66,
Apr 08 21:22:07 osmc hyperiond[652]: "name" : "MyHyperionConfig",
Apr 08 21:22:07 osmc hyperiond[652]: "type" : "ws281x"
Apr 08 21:22:08 osmc hyperiond[652]: }
Apr 08 21:22:07 osmc systemd[1]: hyperion.service: main process exited, code=killed, status=11/SEGV
Apr 08 21:22:07 osmc systemd[1]: Unit hyperion.service entered failed state.
Apr 08 21:22:08 osmc hyperiond[652]: Can't open /dev/mem: Permission denied
Apr 08 21:22:08 osmc systemd[1]: hyperion.service start request repeated too quickly, refusing to start.
Apr 08 21:22:08 osmc systemd[1]: Failed to start Hyperion Systemd service.
Apr 08 21:22:08 osmc systemd[1]: Unit hyperion.service entered failed state.

So hyperion can't start.
Hyperion IS working when I use adalight config. So I assume that the config file for "ws281x" is broken or something.

**Comment by brindosch** [comment]
I´m sorry it is "noticed" if you use the script with usb or ws281x argument it will do the right steps.
https://github.com/tvdzwan/hyperion/wiki/Installation-on-OSMC

osmc has some problems with pwm signal, but not all pwm people are affected
ping @penfold42 
Is this a "known" problem with osmc for ws2811 leds? Do he need another distro?

**Comment by DarkHunterer** [comment]
Ok... So the problem is over. You don't have to bother writting back for my problems which I had till this time... 

I was stubborn and I found the solution after using analyzer... The solution was so obvious...
I am going to write it tomorrow, because I really believe this is going to save much time to people in the future...

However, I still have to check if the ws281x option is working, because now Im using usb->arduino again.

So thank you very much for trying to help me.
Finally Hyperion working... It looks so amazing. Nice job on this project!

**Comment by brindosch** [comment]
Glad that it is working. pwm and osmc is still.. in some way a problem. But we can´t do anything. It is working with Raspbian and OpenELEC.

**Comment by DarkHunterer** [comment]
Ok, so basically what I did was:

1) Read the signal with analyzer. It turned out that the PWM indeed is working. I saw that the '1' level takes like 1.0~1.2 usecond. 

2) So I had to check the documentation of ws2811 if it is right period or it is not. So generally it was not exact with the documentation, not even very close but there should be some flashes from time to time, because some of the bits timing were correct.

3) And the moment I read the documentation was the moment when I realized what I have done... When I mounted the leds I counted physical leds... Not the ws2811 chips... The documentation shows 4 different ways of using ws2811, 5V - 1 led ; 5V - 3 leds parallel ; 12V - 1 led ; 12V - 3 leds parallel.
So basically... I thought I am using 66 leds when I in fact was using 22...

tl;dr:
The ws281x chips can be mounted on strip in 2 different ways : 1 led 1 chip or 3 leds 1 chip. It is really worth checking what is your strip like...

I think it is worth adding to FAQ. Something like "How many leds do I have - that many as you have ws chips"

**Comment by tpmodding** [comment]
hey @DarkHunterer 

that is realy interesting! didnt know that fact in 3)! but i think it is only on ws2811?? 

do you have a stripe that likes like this? http://images.google.de/imgres?imgurl=http%3A%2F%2Fg01.a.alicdn.com%2Fkf%2FHTB1rOvcJpXXXXcoXFXXq6xXFXXXK%2F1M-5M-WS2811-RGB-LED-Strip-12V-SMD-5050-Adressable-Pixel-Strip-Black-or-White-PCB.jpg&imgrefurl=http%3A%2F%2Fwww.aliexpress.com%2Fstore%2Fproduct%2F1m-5m-WS2811-RGB-LED-Strip-12V-SMD-5050-Adressable-Pixel-Strip-Black-or-White-PCB%2F1755004_32461755828.html&h=506&w=700&tbnid=sdt_yU1gLVevJM%3A&docid=_TasIXaPBFoiwM&ei=4VkKV7bQIcrB6AShzpyYAw&tbm=isch&iact=rc&uact=3&dur=1194&page=5&start=86&ndsp=24&ved=0ahUKEwj2xOePm4TMAhXKIJoKHSEnBzM4ZBAzCBUoCTAJ

cause the ws2812b has is chip IN the led! so that it can control only one led...

 i thought hole time there a nearly "same" as shown here:
http://rgb-123.com/wp-content/uploads/2013/08/WS2811vsWS2812b.jpg

how it is possible to check? if you say hyperion you have 2 leds...and make them for example red with "hyperion-remote -c red" than it shoul only flash 2 or 6 leds? right?

**Comment by DarkHunterer** [comment]
@tpmodding This is exactly the strip I bought. And you clearly cannot control each led, I control them by groups of 3.

If I would know that 2812b has chip inside, I would pick it instead :/

You are right, if you would say hyperion that there are only 2 leds and make them flash, 6 first would turn on.
I can check it tomorrow on hyperion, but I did exactly same thing with FastLed library yesterday.

**Comment by tpmodding** [comment]
@DarkHunterer ok thanks! that would be nice to know, to add it to the FAQ.

btw **## i don't recommend** ws2811 or ws2812b  or any PWM led!!! always use leds with data AND CLOCK!!

then you have never those problems :+1: :)

**Comment by penfold42** [comment]
Yuk - sounds like they were being cheap. 

Just so im clear, is each physical led RGB but wired in parallel in groups of 3 to a single 2811 chip ?

**Comment by DarkHunterer** [comment]
@tpmodding Both options have their pros and cons. It is in fact easier to use chips with clock for hyperion but if you have real-time hardware clock on your board it is easier to just use PWM controlled strip. 
I checked the setting with just 2 leds in hyperion.config file. So basicaly I see that the first 2 groups (6 leds) are flashing in desired way. But then there are additonal 6 leds turned on in constant color, and than there is color again. 
Example: hyperion-remote -c RED -d 20000
r r r | r r r| w w w g g g| r r r |r r r| w w w g g g| r r r| r r r| w w w g g g
r-red
w-white
g-green
And it goes till the end of strip. When I change color for GREEN, the w w w g g g scheme stays the same. I suspect it is connected with "Ada"+"3 bytes of information" word, there goes 6 bytes with it = information of 2 colors.
(The colors were not calibrated, so it doesn't have to be green for real)

@penfold42 Yes, there are groups of 3 parallel leds connected to single 2811 chip.
 And they are in fact cheaper but not that much cheaper than ws2812b, I don't understand why the price be any circumstance here. I picked it because there wasn't clear information about the connection or I just didn't understand it.

**Comment by brindosch** [comment]
tpmodding want to say that you may never opend this issue and invest time in fixing - because of this.
Also OSMC guys screwed something up that still make issue and we have no idea what is going wrong.

So just time eating.

**Comment by tpmodding** [comment]
@DarkHunterer the price-thing has nothing to do with you :) but most people try to make his ambilight as cheap as possible and forget the "quality" and than came here with thousend of issues....

but it is "good" to know that you found the problem...

and if you now setup your config knowing 3 leds per chip, does it work without problems?

**Comment by DarkHunterer** [comment]
Yes, hyperion works perfectly now.

So thank you for your time and answers guys, I believe this is solved already :)
You can add a note to FAQ for the future someone like me, who didn't realize that has 3 leds parallely.

Great project.

**Comment by brindosch** [comment]
Good
